src/es/components/prototypes/Shadow.js

```diff
@@ -581,6 +589,101 @@ export const Shadow = (ChosenHTMLElement = HTMLElement) => class Shadow extends
+   * @return {Promise<fetchModulesParams[]>}
+   */
+  fetchModules (fetchModulesParams, hide = true, useController = true) {
+    if (hide) this.hidden = true
+    if (!Array.isArray(fetchModulesParams)) fetchModulesParams = [fetchModulesParams]
+    if (this.isConnected && useController && document.body.hasAttribute(this.getAttribute('fetch-modules') || 'fetch-modules')) {
+      // use: /src/es/components/controllers/fetchHtml/FetchHtml.js instead of fetching here, to use the cache from within the controller
+      return new Promise(
+        /**
+         * setup Promise function
+         *
+         * @param {any} resolve
+         * @return {boolean}
+         */
+        resolve => this.dispatchEvent(new CustomEvent(this.getAttribute('fetch-modules') || 'fetch-modules', {
+          /** @type {import("../controllers/fetchModules/FetchModules.js").fetchModulesEventDetail} */
+          detail: {
+            fetchModulesParams,
+            resolve,
+            node: this
+          },
+          bubbles: true,
+          cancelable: true,
+          composed: true
+        }))
+      ).then(
+        /**
+         * the controller resolving fetch-modules will return with its fetchHtml results
+         *
+         * @param {fetchModulesParams[]} resultFetchModulesParams
+         * @return {fetchModulesParams[]}
+         */
+        resultFetchModulesParams => {
+          if (hide) this.hidden = false
+          return resultFetchModulesParams
+        }
+      )
+    } else {
+      return Promise.all(fetchModulesParams.map(
+        /**
+         * fetch each fetchHTMLParam.paths and return the promise
+         *
+         * @param {fetchModulesParams} fetchModulesParam
+         * @return {Promise<fetchModulesParams>}
+         */
+        fetchModulesParam => (import(fetchModulesParam.path).then(
+          /**
+           * return the paths with the response.text or an Error
+           *
+           * @param {any} module
+           * @return {fetchModulesParams}
+           */
+          module => {
+            let constructorClass = module.default || module
+            if (typeof constructorClass === 'object') constructorClass = constructorClass[Object.keys(constructorClass)[0]]()
+            if (!customElements.get(fetchModulesParam.name)) customElements.define(fetchModulesParam.name, constructorClass)
+            fetchModulesParam.constructorClass = constructorClass
+            return fetchModulesParam
+          }
+        )).catch(
+          /**
+           * Return the paths with the attached error
+           *
+           * @param {string} error
+           * @return {fetchModulesParams}
+           */
+          error => {
+            if (hide) this.hidden = false
+            error = `${fetchModulesParam.path} ${error}!!!`
+            // @ts-ignore
+            return { ...fetchModulesParam, error: (this.html = console.error(error, this) || `<code style="color: red;">${error}</code>`) }
```

Comment:
`this.html` may not exist. Probably `this.innerHTML` was meant? 🤔

Reply:
https://github.com/mits-gossau/web-components-toolbox/blob/master/src/es/components/prototypes/Shadow.js#L598